hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -176,9 +176,7 @@ public boolean isEqual(final Object x, final Object y) throws HibernateException
 		if ( x == y ) {
 			return true;
 		}
-		if ( x == null || y == null ) {
```

Comment:
Yes, we did that on purpose as we think it is a bug in the current behavior.

With the current behavior, when you persist your entity with an empty embedded and you get it back the embedded is null. So we think in this case, the empty embedded and null should be considered equal.

I can change it if you think it's better to keep the old behavior.